### PR TITLE
Fix package:shelf_router to ensure GET handlers also respond to HEAD …

### LINKED ISSUE #34: Using `router.get` should add a default handler for `router.head`
We should add default handlers for `HEAD` when we have handlers for `GET`.
Probably we can just serve the response from the `GET` handler if there is no other handler registered that handles the `HEAD` request.

```dart
// when writing
router.get('/', myHandler);
// we should implicitly add:
router.head('/', myHandler); // if and only if, no other registered handler matches the HEAD request.
```

@isoos, does this sound like a reasonable solution to you? I think this might solve some problems for us in `pub.dev`.

### PR DESCRIPTION
**Pull request by jonasfj**
…requests

Fixes #34

### CODE REVIEW

File: shelf_router/lib/src/router.dart [hunk 1]
```diff
@@ -84,6 +95,11 @@ class Router {
     }
     verb = verb.toUpperCase();
 
+    if (verb == 'GET') {
```
Comment (isoos) [inline shelf_router/lib/src/router.dart]: What happens if you specify both `HEAD` and `GET`?

Comment (jonasfj) [reply shelf_router/lib/src/router.dart]: As long as you specify the `HEAD` handler before the `GET` handler you'll be fine.

Comment (isoos) [reply shelf_router/lib/src/router.dart]: This should be documented somewhere, otherwise this is good to go.

Comment (jonasfj) [reply shelf_router/lib/src/router.dart]: Good point, I documented it for `Router.get` but we should also document it here.

**Pull request merged**